drivers/gpio/gpio_altera_pio.c

```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright (c) 2022, Intel Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT altr_pio
+
+/**
+ * @file
+ * @brief PIO driver for Intel FPGA PIO Core IP
+ * Reference : Embedded Peripherals IP User Guide : 27. PIO Core
+ */
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/printk.h>
+#include <inttypes.h>
+#include "gpio_utils.h"
+#include <string.h>
+#include <stdbool.h>
```

Comment:
Looks like your list of includes could use a bit of trimming (e.g. printk.h is unused, several of the others look to be unused as well).